pkg/virt-controller/watch/vmi.go

```diff
@@ -2127,10 +2127,13 @@ func (c *VMIController) updateVolumeStatus(vmi *virtv1.VirtualMachineInstance, v
 }
 
 func (c *VMIController) getFilesystemOverhead(pvc *k8sv1.PersistentVolumeClaim) (cdiv1.Percent, error) {
-	_, cdiExists, _ := c.cdiInformer.GetStore().GetByKey("cdi")
-	if !cdiExists {
+	// To avoid conflicts, we only allow having one CDI instance
+	if cdiInstances := len(c.cdiInformer.GetStore().List()); cdiInstances > 1 {
+		return "0", fmt.Errorf("more than one CDI resource detected: %v", cdiInstances)
+	} else if cdiInstances == 0 {
```

Comment:
I know I already commented that in a different PR but no need the else if you return in the first if :)